### PR TITLE
docker test: use bazel.FindBinary() to find docker-fsnotify

### PR DESCRIPTION
**Pull request by jlinder**
Before: the binary was being found manually.

Why: the old method of finding the binary was not correct. It may have
also been broken bay a bazel builder image update which then caused the
docker image test to start flaking.

Now: use the bazel provided way of finding a dependant binary:
`bazel.FindBinary()`.

Release note: None

### CODE REVIEW

File: pkg/testutils/docker/single_node_docker_test.go [hunk 1]
```diff
@@ -93,7 +96,7 @@ func TestSingleNodeDocker(t *testing.T) {
 				},
 				volSetting: []string{
 					fmt.Sprintf("%s/testdata/single-node-test/docker-entrypoint-initdb.d/:/docker-entrypoint-initdb.d", pwd),
```
Comment (rickystewart) [inline pkg/testutils/docker/single_node_docker_test.go:98]: Drive-by: It is also preferred to use `testutils.TestDataPath()` rather than literally referring to `testdata`. [#71928](https://github.com/cockroachdb/cockroach/issues/71928)

**Review by ZhouXing19 — COMMENTED** [review]
Thanks this is much neater! 
LGTM modulo the import of `"github.com/cockroachdb/cockroach/pkg/build/bazel"` is failing the lint test.
Also `pkg/testutils/docker/BUILD.bazel` needs to be updated with the new import.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/79089#-:-MzVOg_i4FWTCAHv8uHq:b-gubtdm)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jlinder and @rickystewart)
